ueberdosis/tiptap
# Remove tippy.js and replace with Floating UI

## Changes Overview
This PR completely removes tippy.js from all Tiptap packages and replaces the implementation with [Floating UI](https://www.floating-ui.com).

Tippy was already outdated and often caused rendering issues for example with React or conditional rendering. The new implement Floating UI implementation now can be conditionally rendered, customized and adjusted.

## Implementation Approach
Instead of letting Tippy handle the element creation and mounting, we now do this on our own and use Floating UI to compute the position of the floating element using [virtual elements](https://floating-ui.com/docs/virtual-elements).

The positions are calculated from the cursor or selection position.

Instead of `tippyOptions` we now pass down an `option` object which can be used to control the Floating UI middlewares like `shift`, `flip` or `offset`.

## Testing Done
I updated all tests that were previously using tippy and made sure all run successfully.

## Verification Steps

1. Try out this change manually on our demos locally
2. Check the test runs

## Additional Notes

Breaking change incoming. 😬

## Checklist
- [x] Add documentation in https://github.com/ueberdosis/tiptap-docs (See https://github.com/ueberdosis/tiptap-docs/pull/16)

## Related Issues

- #3170 (now we bundle Floating UI into the project - we could optionally use peerDependencies here, @nperez0111, @svenadlung  what do you think?)
